scripts/bump_extension_version.js

```diff
@@ -0,0 +1,153 @@
+const fs = require('fs')
+const path = require('path')
+const semVer = require('semver')
+
+function isMinorRelease(
+  prismaVersion,
+) {
+  const tokens = prismaVersion.split('.')
+  if (tokens.length !== 3) {
+    throw new Error(
+      `Version ${prismaVersion} must have 3 tokens separated by "." character.`
+    )
+  }
+  return tokens[2] === '0'
+}
+
+function currentExtensionVersion({
+  branch_channel
+}) {
+  switch (branch_channel) {
+    case 'master':
+    case 'dev':
+      return readFile({ fileName: "extension_insider" })
+    case 'latest':
+      return readFile({ fileName: "extension_stable" })
+    case 'patch-dev':
+      return readFile({ fileName: "extension_patch" })
+    default:
+      if (branch_channel.endsWith('.x')) {
+        return readFile({ fileName: "extension_patch" })
+      }
+  }
+}
+
+function stripPreReleaseText(version) {
+  return version.replace('-dev', '')
+}
+
+function nextVersion({
+  currentVersion,
+  branch_channel,
+}) {
+  const prisma_latest = readFile({ fileName: 'prisma_latest' })
+  const prisma_dev = stripPreReleaseText(readFile({ fileName: 'prisma_dev' }))
+  const prisma_patch = stripPreReleaseText(readFile({ fileName: 'prisma_patch' }))
+
+  const currentVersionTokens = currentVersion.split('.')
+  const prisma_dev_tokens = prisma_dev.split('.')
+  const prisma_latest_tokens = prisma_latest.split('.')
+
+  switch (branch_channel) {
+    case 'master':
+    case 'dev':
+      // Prisma CLI new dev version
+      if (prisma_dev_tokens[2] != currentVersionTokens[1]) {
+        // first new release after patch bump --> extensionVersion from x.y.z to x.(y+1).1
+        console.log("First new release after patch bump.")
+        let bumpMinor = semVer.inc(currentVersion, 'minor').split('.')
+        return `${bumpMinor[0]}.${bumpMinor[1]}.1`
+      }
+      if (prisma_latest_tokens[1] == currentVersionTokens[0]) {
+        // first new release after stable minor bump --> extensionVersion from x.y.z to (x+1).0.1
+        console.log("First new release after stable minor bump.")
+        let bumpMajor = semVer.inc(currentVersion, 'major').split('.')
+        return bumpMajor[0] + '.0.1'
+      }
+      return semVer.inc(currentVersion, 'patch')
+    case 'latest':
+      // Prisma CLI new latest version
+      if (isMinorRelease(prisma_latest)) {
+        return prisma_latest
+      }
+      return semVer.inc(currentVersion, 'patch')
+    case 'patch-dev':
+      const derivedVersion = getDerivedExtensionVersion(stripPreReleaseText(prisma_patch))
+      if (prisma_patch_tokens[0] !== currentVersion[0]) {
+        return derivedVersion
+      }
+
+      return semVer.inc(currentVersion, 'patch')
+    default:
+      if (branch_channel.endsWith('.x')) {
+        // extension only new patch update
+        if (prisma_latest_tokens[0] !== currentVersionTokens[0]) {
+          return `${prisma_latest_tokens[0]}.1.1`
+        }
+        return semVer.inc(currentVersion, 'patch')
+      }
```

Comment:
Same, probably a good idea to throw an error in the else block, in case this is called incorrectly.